### PR TITLE
MSP telemetry and OSD support

### PR DESCRIPTION
**Pull request by chris1seto**
**Overview**
This PR adds support for streaming telemetry over the MSP protocol. For now, it's specifically targeted to supporting the OSD functionality in the DJI Air Unit. 

**msp_osd module functionality**
The new module implemented in this PR consumes data from uORB, and then immediately packs it into MSP messages to be transmitted over serial.

**What is MSP?**
MSP stands of MultiWii Serial Protocol. While the name has its roots in the outdated MultiWii project, MSP is used today in a variety of modern UAV applications for exchange of sensor data. The MSP protocol isn't concretely documented, but there are full implementation examples which can be found in Ardupilot, Betaflight, inav, etc. MSP is a simple serial based bytestream protocol. MSP exchanged can be RX only, TX only, or both request/response RXTX. For the purposes of this PR, we only TX. 

**What could or should be done additionally in this PR?**
* CPP/PX4 project best practices (eg, cpp-ify/refactor MSPV1 lib)
* Overall project integration  (eg, how to set target serial port? Autostart, etc)
* More sophisticated handling in how the various uORB messages are merged into MSP messages (eg, how to deal with invalid uORB message content from messages which have not been published to since start of msp_osd runtime)
* Customization in OSd element placement on the actual screen (eg, perhaps a PX4 param to set X/Y locations of each element) 

**MSP resources**
https://github.com/iNavFlight/inav/wiki/MSP-V2
https://github.com/d3ngit/djihdfpv_mavlink_to_msp_V2
https://ardupilot.org/plane/docs/common-msp-osd-overview.html
https://github.com/ArduPilot/ardupilot/tree/master/libraries/AP_MSP
https://github.com/betaflight/betaflight/tree/master/src/main/msp
https://github.com/iNavFlight/inav/tree/master/src/main/msp

### CODE REVIEW

File: src/modules/msp_osd/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+############################################################################
+#
+#   Copyright (c) 2018 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/modules/msp_osd/CMakeLists.txt:3]:
```suggestion
#   Copyright (c) 2022 PX4 Development Team. All rights reserved.
```

File: src/modules/msp_osd/msp_osd.hpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2018 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/modules/msp_osd/msp_osd.hpp:3]:
```suggestion
 *   Copyright (c) 2022 PX4 Development Team. All rights reserved.
```

**Comment by dagar** [comment]
I resolved the merge conflicts, fixed compile failures (vehicle_gps_position rename), added missing copyright headers, and moved this to drivers/osd/msp_osd.

This still isn't autostarting via parameter so it should be harmless to merge and we can continue in https://github.com/PX4/PX4-Autopilot/pull/20079.

**Comment by hamishwillee** [comment]
This is cool. @chris1seto Plan for docs when this goes in?

**Comment by sastel** [comment]
Hey @romain-chiap there is a similar issue with the `msp_osd` driver and the Walksnail Avatar system. See [this discussion](https://discuss.px4.io/t/msp-osd-support-for-walksnail-vtx-and-avatar-goggles/40134/8). I think the driver was written for a DJI Air system prior to the O3 Air Unit, and doesn't work with the O3 (and Walksnail).

**Comment by Li-Tianming** [comment]
I captured the MSPOSD data of PX4 and Ardupilot, and they look very different, 
The MSPOSD of Ardupilotis working properly, perhaps the MSPOSD protocol of PX4 is outdated?

 Girl in a jacket

 Girl in a jacket

**Pull request merged**